### ISSUE TITLE
Example/Documentation update [TFJS expected axis NaN for dense1]

### ISSUE THREAD

**Issue by carlos-r-l-rodrigues** [Pravez/ReImproveJS#2]
It seems that the model must goes inside the Agent config:
`
model.compile({loss: 'categoricalCrossentropy', optimizer: 'sgd'})
const agentConfig = {
    model: model,
    memorySize: 5000,
    batchSize: 128,
    temporalWindow: temporalWindow
};`

`await academy.step([
        {teacherName: teacher, inputs: inputs}
    ]);`
The parameters in the interface AcademyStepInput are "teacherName" and "agentsInput".

**Comment by Pravez** [comment]
Indeed, thank you for noticing it ! Corrected.

**Comment by carlos-r-l-rodrigues** [comment]
The model should be like this:
`const agentConfig = {
	model: model,
	agentConfig: {
		memorySize: 1000,
		batchSize: 10,
		temporalWindow: temporalWindow
	}
};`

Also, the academy.step should be:
`await academy.step([
				{ teacherName: teacher, agentsInput: inputs }
			])`

I'm also debugging another issue with tensorflow.js but I haven't figured out why it's happen.
Error: Input 0 is incompatible with layer dense_Dense1: expected axis NaN of input shape to have value

**Comment by Pravez** [comment]
Thank you once more, I forgot the second part of the correction...

Can you provide an exemple to reproduce your bug ?

**Comment by carlos-r-l-rodrigues** [comment]
Please ignore the error, it was caused by another third part code overriding the Object.prototype.clone.
At some point it was changing the type of the objects used by ts.js and failing in the assert.

Thanks for this awesome framework 👍
